### PR TITLE
Async workflow request consumer manager in worker

### PR DESCRIPTION
**Pull request by taylanisikdemir**
**What changed?**
This PR adds implementation of async workflow request consumer manager which starts/stops consumers periodically based on domain configs.

**How did you test it?**
Added new unit tests. Integration tests will be added in follow up PRs.

### CODE REVIEW

File: common/metrics/defs.go [hunk 1]
```diff
@@ -1341,6 +1341,8 @@ const (
 	CheckDataCorruptionWorkflowScope
 	// ESAnalyzerScope is scope used by ElasticSearch Analyzer (esanalyzer) workflow
 	ESAnalyzerScope
+	// AsyncWorkflowConsumerScope is scope used by async workflow consumer
+	AsyncWorkflowConsumerScope
```
Comment (Shaddoll) [inline common/metrics/defs.go:1345]: the definition is missing

File: service/worker/asyncworkflow/async_workflow_consumer_manager.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+		cfg := domain.GetConfig().AsyncWorkflowConfig
+		queue, err := c.getQueue(cfg)
+		if err != nil {
+			c.logger.Error("Failed to get queue", tag.Error(err), tag.WorkflowDomainName(domain.GetInfo().Name))
+			continue
+		}
+
+		if !cfg.Enabled {
+			// lready running active consumers for such queues will be stopped in the next loop
```
Comment (Shaddoll) [inline service/worker/asyncworkflow/async_workflow_consumer_manager.go:165]: typo: already

File: common/metrics/defs.go (hunk 1 above)
Comment (taylanisikdemir) [reply common/metrics/defs.go:1345]: thanks for the catch

File: service/worker/asyncworkflow/async_workflow_consumer_manager.go
Comment (Shaddoll) [inline service/worker/asyncworkflow/async_workflow_consumer_manager.go:1]: We're not processing the messages read from consumers. We only start and stop consumers in the consumer manager. Are you planning to introduce another component to processing the messages?

File: service/worker/service.go
Comment (Shaddoll) [inline service/worker/service.go:1]: It seems that the stop processes for several components of worker are not triggered. But we can fix it in a separate PR.

File: service/worker/asyncworkflow/async_workflow_consumer_manager.go
Comment (taylanisikdemir) [reply service/worker/asyncworkflow/async_workflow_consumer_manager.go:1]: yes. left it out to keep PR small

File: service/worker/service.go
Comment (taylanisikdemir) [reply service/worker/service.go:1]: I was going to make all of those components work like new asyn wf consumer but some of those components are not stoppable so require more work to get there. At least the new component shows how it should be done. It should be easy to follow same pattern when we decide to refactor rest.

**Review by Shaddoll — APPROVED** [review]

**Pull request merged**